src/overlays/actors/ovl_Fuusen/ac_fuusen.c

```diff
@@ -22,34 +36,437 @@ ActorProfile Fuusen_Profile = {
+            if (!(this->unk_18C & 2)) {
+                this->actor.shape.rot.z = 500;
+            } else {
+                this->actor.shape.rot.z = 0;
+            }
+            this->unk_18C++;
+        } else {
+            this->actor.shape.rot.z = 0;
+            this->unk_18C = 0;
+        }
+    }
+}
+
+void aFSN_escape(Actor* thisx, UNUSED Game_Play* game_play) {
+    Fuusen* this = THIS;
+    f32 balloonGroundY = mCoBG_GetBalloonGroundY(&this->actor.world.pos);
+
+    if (this->unk_18C == 0) {
+        xyz_t pos;
+        s32 utX;
+        s32 utZ;
+
+        mFI_Wpos2UtCenterWpos(&pos, this->actor.world.pos);
+        if ((common_data.clip.unk_074 != NULL) && (common_data.clip.unk_074->unk_38 != 0) &&
+            mFI_Wpos2UtNum(&utX, &utZ, pos)) {
+            common_data.clip.unk_074->unk_30(9500, utX, utZ, 1);
+            this->unk_18C = 1;
+        }
+    }
+    if (this->unk_18C == 1) {
+        if (this->actor.world.pos.y > (balloonGroundY + 500.0f)) {
+            Actor_delete(&this->actor);
+        }
+    }
+}
+
+FuusenInitFunc init_proc[4] = {
+    aFSN_birth_init,
+    aFSN_moving_init,
+    aFSN_wood_stop_init,
+    aFSN_escape_init,
+};
+
+FuusenActionFunc act_proc[4] = {
+    aFSN_birth,
+    aFSN_moving,
+    aFSN_wood_stop,
+    aFSN_escape,
+};
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Fuusen/ac_fuusen/aFSN_actor_dt.s")
+void aFSN_setupAction(Fuusen* this, s32 processIndex, Game_Play* game_play) {
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Fuusen/ac_fuusen/func_80AAD270_jp.s")
+    this->processNum = processIndex;
+    this->process = act_proc[processIndex];
+    init_proc[processIndex](this, game_play);
+}
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Fuusen/ac_fuusen/func_80AAD490_jp.s")
+// Original name unknown.
+void func_80AADF10_jp(Fuusen* this, Game_Play* game_play) {
+    UNUSED UNK_TYPE1 pad[0x4];
+    void* segment;
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Fuusen/ac_fuusen/func_80AAD4A0_jp.s")
+    segment = game_play->objectExchangeBank.status[this->actor.unk_026].segment;
+    if (this->segment != segment) {
+        SkeletonInfoR* skeletonInfo = &this->skeletonInfo;
+        skeletonInfo->skeleton = Lib_SegmentedToVirtual(&cKF_bs_r_act_balloon);
+        skeletonInfo->animation = Lib_SegmentedToVirtual(&cKF_ba_r_act_balloon);
+        this->segment = segment;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Fuusen/ac_fuusen/func_80AAD4DC_jp.s")
+void aFSN_actor_move(Actor* thisx, Game_Play* game_play) {
+    Fuusen* this = THIS;
+    UNUSED UNK_TYPE1 pad[0x8];
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Fuusen/ac_fuusen/func_80AAD580_jp.s")
+    func_80AADF10_jp(this, game_play);
+    if (this->birthTimer == 0) {
+        Actor_position_moveF(&this->actor);
+    } else if (this->birthTimer > 0) {
+        this->birthTimer--;
+    }
+    if (D_80106E10_jp != 0) {
```

Comment:
`fuusen_DEBUG_mode_flag`?

Reply:
not sure how to confirm this is the case, but will change to that